### ISSUE TITLE
InfoDomainRequest error: DOMXPath::query(): Undefined namespace prefix

### ISSUE THREAD

**Issue by jcerdan** [struzik-vladislav/epp-client#4]
Hi, 
I get this error on InfoDomainRequest

`
  DOMXPath::query(): Undefined namespace prefix

  at vendor/struzik-vladislav/epp-client/src/Response/AbstractResponse.php:55
     51▕      * {@inheritdoc}
     52▕      */
     53▕     public function get($xpathQuery, \DOMNode $contextNode = null): \DOMNodeList
     54▕     {
  ➜  55▕         return $this->xpath->query($xpathQuery, $contextNode);
     56▕     }
     57▕ 
     58▕     /**
     59▕      * {@inheritdoc}
`

Have you ever met this problem?

**Comment by jcerdan** [comment]
The server returns this id, version, languages and NamespaceURI:

ID: CORE EPP Server (tango-1.118 r20295)
Version: 1.0
Languages:  en
Namespaces:
urn:ietf:params:xml:ns:epp:maintenance-1.0
urn:ietf:params:xml:ns:host-1.0
urn:ietf:params:xml:ns:domain-1.0
urn:ietf:params:xml:ns:contact-1.0

And this is the answer for info request

```
<?xml version=\"1.0\" encoding=\"UTF-8\"?>
<epp xmlns=\"urn:ietf:params:xml:ns:epp-1.0\">
<response>
<result code=\"1000\">
<msg>Command completed successfully</msg>
</result>
<resData>
<infData xmlns=\"urn:ietf:params:xml:ns:domain-1.0\">
<name>*******</name>
<roid>D2121-AD</roid>
<status s=\"clientTransferProhibited\"/>
<registrant>**********</registrant>
<contact type=\"admin\">**********</contact>
<ns><hostObj>**********</hostObj><hostObj>**********</hostObj><hostObj>**********</hostObj></ns>
<clID>*****</clID>
<crID>REGISTRY</crID>
<crDate>****************</crDate>
<upDate>***************</upDate>
<exDate>****************</exDate>
<trDate>****************</trDate>
<authInfo><pw>****************</pw></authInfo>
</infData>
</resData><extension>
<infData xmlns=\"http://xmlns.corenic.net/epp/privacy-1.0\">
<disclosure><legal/></disclosure>
</infData></extension>
<trID>
<clTRID>I
</clTRID><svTRID>******************</svTRID>
</trID>
</response>
</epp>
```

I have formatted it a bit so it's a little more readable and hidden info with '*'.

I connect to the server with the following namespaces

```
$client->getNamespaceCollection()->offsetSet(NamespaceCollection::NS_NAME_ROOT, 'urn:ietf:params:xml:ns:epp-1.0');
$client->getNamespaceCollection()->offsetSet(NamespaceCollection::NS_NAME_CONTACT, 'urn:ietf:params:xml:ns:contact-1.0');
$client->getNamespaceCollection()->offsetSet(NamespaceCollection::NS_NAME_HOST, 'urn:ietf:params:xml:ns:host-1.0');
$client->getNamespaceCollection()->offsetSet(NamespaceCollection::NS_NAME_DOMAIN, 'urn:ietf:params:xml:ns:domain-1.0');
```

regards

**Comment by struzik-vladislav** [comment]
Hi, @jcerdan 
Please write the XPath query that you pass when you get this error?

**Comment by struzik-vladislav** [comment]
The response looks wrong. Namespaces do not have prefixes like domain, contact, etc. It is incompatible with [RFC 5730](https://tools.ietf.org/html/rfc5730) - Extensible Provisioning Protocol (EPP), [RFC 3731](https://tools.ietf.org/html/rfc3731) - Extensible Provisioning Protocol (EPP) Domain Name Mapping.

**Comment by jcerdan** [comment]
Hi @struzik-vladislav ,

I have contacted the server assistance. they say their response complies with RFC 5730 as it says:

"EPP uses XML namespaces to provide an extensible object management
framework and to identify schemas required for XML instance parsing
and validation. These namespaces and schema definitions are used to
identify both the base protocol schema and the schemas for managed
objects. The XML namespace prefixes used in examples (such as the
string "foo" in "xmlns:foo") are solely for illustrative purposes. A
conforming implementation MUST NOT require the use of these or any
other specific namespace prefixes."

Looks like it will be a no issue for me and I will have to look for another library. What a pity as yours was really clear and easy to use :-(

Thanks a lot for your assistance

**Comment by struzik-vladislav** [comment]
Hi @jcerdan 
This bug has been fixed. The fix is released in v2.5.1

regards
